### PR TITLE
c2c: refactor eventStream to track batch in seperate streamEventBatcher

### PR DESCRIPTION
**Pull request by msbutler**
This is a small refactor cleans up the streamLoop() in the eventStream and makes it easier to emit spanConfig protos for zone config replication.

Epic: none

Release note: None

### CODE REVIEW

**Comment by msbutler** [comment]
unrelated flake

File: pkg/ccl/streamingccl/streamproducer/batch_manager.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+package streamproducer
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+)
+
+type batchManager struct {
```
Comment (lidorcarmel) [inline pkg/ccl/streamingccl/streamproducer/batch_manager.go:17]: add a batch_manager_test.go file?

Comment (msbutler) [reply pkg/ccl/streamingccl/streamproducer/batch_manager.go:17]: good callout. done.

Comment (stevendanna) [inline pkg/ccl/streamingccl/streamproducer/batch_manager.go:17]: When I see `manager` I typically think of something that oversees a collection of a thing. But, let's not bikeshed the name.

File: pkg/ccl/streamingccl/streamproducer/event_stream.go [hunk 2]
```diff
@@ -361,84 +361,59 @@ func (p *checkpointPacer) shouldCheckpoint(
...
 	// If the sst span exceeds boundaries of the watched spans,
 	// we trim the sst data to avoid sending unnecessary data.
 	// TODO(casper): add metrics to track number of SSTs, and number of ssts
 	// that are not inside the boundaries (and possible count+size of kvs in such ssts).
-	size := 0
+	//
 	// Extract the received SST to only contain data within the boundaries of
 	// matching registered span. Execute the specified operations on each MVCC
 	// key value and each MVCCRangeKey value in the trimmed SSTable.
 	if err := replicationutils.ScanSST(sst, registeredSpan,
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamproducer/event_stream.go:382]: Existing code, but if you wanted: `return replicationutils.ScanSST(sst, registeredSpan,...`

File: pkg/ccl/streamingccl/replicationutils/utils.go [hunk 3]
```diff
@@ -39,6 +39,7 @@ import (
 func ScanSST(
 	sst *kvpb.RangeFeedSSTable,
 	scanWithin roachpb.Span,
+	// TODO (msbutler): I think we can use a roachpb.kv instead, avoiding EncodeDecode roundtrip.
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/replicationutils/utils.go:42]: Interesting.

**Review by stevendanna — APPROVED** [review]
Thanks for doing these kinda cleanups.

I left some nits, but they are mostly just notes for the next time we touch this code.

File: pkg/ccl/streamingccl/streamproducer/batch_manager.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/streamingccl/streamproducer/batch_manager.go]: refactored to `batcher`. we can further bikeshed after this pr merges :)

File: pkg/ccl/streamingccl/streamproducer/event_stream.go (hunk 2 above)
Comment (msbutler) [reply pkg/ccl/streamingccl/streamproducer/event_stream.go]: done.

**Comment by msbutler** [comment]
bors r=stevendanna

**Pull request merged**